cmd/crdvalidator/README.md

```diff
@@ -0,0 +1,116 @@
+# crdvalidator
+
+## Summary
+A part of the core value proposition for RukPak is the safe upgrade of a [CustomResourceDefinition](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/) (`CRD`). As a result, this repository comes equipped with a "CRD Validator Webhook", `crdvalidator`, that will validate specified `CRD` upgrade events. This protects your CRD upgrades from having potentially dangerous effects such as data loss. 
+
+
+### How to use
+
+The CRD Validator webhook will ensure that any CRD upgrade is safe as long as the incoming CRD has the `core.rukpak.io/enable-safe-upgrade-validation` annotation set to `enabled` or if it is owned by a RukPak provisioner.
```

Comment:
I wonder if `safe-upgrade-validation` is misleading. I'm sure there are upgrades this webhook would deem safe that in fact are not safe (e.g. a conversion webhook in the CRD does something wrong -- it would be impossible for us to tell).

Should we have explicit annotations for each of the checks that we _are_ doing? For example: 
- `core.rukpak.io/block-removal-of-inuse-stored-version: true`
- `core.rukpak.io/block-schema-tightening: true`

Reply:
Seems reasonable, since we're still on the "breaking changes are what we do" train :)

Reply:
Created #289, and made a note about the uncertainly around whether it makes sense to expose this kind of configuration behavior. These kinds of configuration knobs appear to be appropriate for upstream environments, but potentially problematic in downstream environments.